app/src/test/java/org/mozilla/fenix/components/PrivateShortcutCreateManagerTest.kt

```diff
@@ -0,0 +1,107 @@
+
+class PrivateShortcutCreateManagerTest {
+
+    @Test
+    fun `when createPrivateShortcut is called and Pin shortcut is supported then requestPinShortcut should be called`(){
+        val context = mockkClass(Context::class)
+
+        val privateShortcutCreateManager = spyk(PrivateShortcutCreateManager)
+
+        mockkConstructor(Intent::class)
+
+        val homeActivityIntent = getMockedHomeActivityIntent()
+
+        val shortcut = getMockedShortcutInfoCompat(context, homeActivityIntent)
+
+        val intentSender = getMockedIntentSender(context)
+
+        mockShortcutCompatManagerAndMethods(context, shortcut, intentSender)
+
+        privateShortcutCreateManager.createPrivateShortcut(context)
+
+        verify (exactly = 1) { ShortcutManagerCompat.isRequestPinShortcutSupported(context) }
+        verify (exactly = 1) { ShortcutManagerCompat.requestPinShortcut(context,shortcut,intentSender) }
+    }
+
+
+
+    private fun getMockedShortcutInfoCompat(context: Context, intent:Intent) : ShortcutInfoCompat {
+        val mockIcon = mockkClass(IconCompat::class)
+        mockkStatic(IconCompat::class)
+        every { IconCompat.createWithResource(context, R.mipmap.ic_launcher_private_round) } returns mockIcon
+
+        val mockUUID = mockkClass(UUID::class)
+        mockkStatic(UUID::class)
+        every { UUID.randomUUID() } returns mockUUID
+        every { mockUUID.toString() } returns "fake-uuid"
+
+        every { context.getString(R.string.app_name) } returns "Firefox Preview"
+        every { context.getString(R.string.app_name_private_5, "Firefox Preview") } returns "Private Firefox Preview"
+
+        val mockBuilder = mockkClass(ShortcutInfoCompat.Builder::class)
+        mockkConstructor(ShortcutInfoCompat.Builder::class)
+        every { anyConstructed<ShortcutInfoCompat.Builder>().setShortLabel("Private Firefox Preview") } returns mockBuilder
+        every { mockBuilder.setLongLabel("Private Firefox Preview") } returns mockBuilder
+        every { mockBuilder.setIcon(mockIcon) } returns mockBuilder
+        every { mockBuilder.setIntent(intent) } returns mockBuilder
+
+        val mockShortcutInfoCompat = mockkClass(ShortcutInfoCompat::class)
+        every { mockBuilder.build() } returns mockShortcutInfoCompat
+
+        return mockShortcutInfoCompat
+    }
+
+    private fun getMockedHomeActivityIntent() : Intent {
+        val homeActivityIntent =  mockkClass(Intent::class)
+        every { anyConstructed<Intent>().setAction(Intent.ACTION_VIEW) } returns homeActivityIntent
+        every { homeActivityIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK) } returns homeActivityIntent
+        every { homeActivityIntent.putExtra(HomeActivity.PRIVATE_BROWSING_MODE, true) } returns homeActivityIntent
+        every { homeActivityIntent.putExtra(
+            HomeActivity.OPEN_TO_SEARCH,
+            StartSearchIntentProcessor.PRIVATE_BROWSING_PINNED_SHORTCUT
+        ) } returns homeActivityIntent
```

Comment:
This is a good example of why this approach of writing the test is not optimal.
Here we ensure that **if** the code under test calls this methods they will get the intent that we'll match against in the test.
But the test will pass even if the production code does not call these methods (which would be bad and we'd like to protect against such a regression) because in all cases we return from here an empty intent.
Ideally we want to ensure the Intent used in the production code is exactly the one we want and for this we need to check for all intended properties.